tests/kernel/pending/src/pend.c

```diff
@@ -307,8 +309,7 @@ void task_monitor(void)
 	    (task_high_state != FIFO_TEST_START + 2) ||
 	    (coop_low_state != FIFO_TEST_START + 3) ||
 	    (coop_high_state != FIFO_TEST_START + 4)) {
-		TC_ERROR("**** Threads timed-out in unexpected order\n");
-		goto error;
+		zassert_true(0, "**** Threads timed-out in unexpected order");
```

Comment:
why you need to use zassert_* as you are anyway doing if block checking and all...this will add unnecessary if check for 0.

you can do something like:
 zassert_true( (task_high_state != FIFO_TEST_START + 2) ||
                       (coop_low_state != FIFO_TEST_START + 3) ||
                        (coop_high_state != FIFO_TEST_START + 4),
                         "**** Threads timed-out in unexpected order");

you do the same for all other cases a well.

Reply:
Done